crates/oxc_linter/src/rules/jest/prefer_mock_promise_shorthand.rs

```diff
@@ -0,0 +1,441 @@
+                aVariable.mockReturnValueOnce(
+                    Promise.reject(
+                        new Error('oh noes!')
+                    )
+                )
+            ",
+            None,
+        ),
+        ("jest.fn().mockReturnValue(Promise.resolve(42), xyz)", None),
+        ("jest.fn().mockImplementation(() => Promise.reject(42), xyz)", None),
+        ("aVariable.mockReturnValueOnce(Promise.resolve(42, xyz))", None),
+        ("aVariable.mockReturnValueOnce(Promise.resolve())", None),
+        (
+            "jest.spyOn(fs, \"readFile\").mockReturnValue(Promise.reject(new Error(\"oh noes!\")))",
+            None,
+        ),
+    ];
+
+    let fix = vec![
+        (
+            "jest.fn().mockImplementation(() => Promise.resolve(42))",
+            "jest.fn().mockResolvedValue(42)",
+            None,
+        ),
+        (
+            "jest.fn().mockImplementation(() => Promise.reject(42))",
+            "jest.fn().mockRejectedValue(42)",
+            None,
+        ),
+        (
+            "aVariable.mockImplementation(() => Promise.resolve(42))",
+            "aVariable.mockResolvedValue(42)",
+            None,
+        ),
+        (
+            "aVariable.mockImplementation(() => {
+                return Promise.resolve(42);
+            });",
+            "aVariable.mockResolvedValue(42);",
+            None,
+        ),
+        (
+            "aVariable.mockImplementation(() => Promise.reject(42))",
+            "aVariable.mockRejectedValue(42)",
+            None,
+        ),
+        (
+            "aVariable.mockImplementation(() => Promise.reject(42),)",
+            "aVariable.mockRejectedValue(42,)",
+            None,
+        ),
+        (
+            "aVariable.mockImplementationOnce(() => Promise.resolve(42))",
+            "aVariable.mockResolvedValueOnce(42)",
+            None,
+        ),
+        (
+            "aVariable.mockImplementationOnce(() => Promise.reject(42))",
+            "aVariable.mockRejectedValueOnce(42)",
+            None,
+        ),
+        ("jest.fn().mockReturnValue(Promise.resolve(42))", "jest.fn().mockResolvedValue(42)", None),
+        ("jest.fn().mockReturnValue(Promise.reject(42))", "jest.fn().mockRejectedValue(42)", None),
+        ("aVariable.mockReturnValue(Promise.resolve(42))", "aVariable.mockResolvedValue(42)", None),
+        ("aVariable.mockReturnValue(Promise.reject(42))", "aVariable.mockRejectedValue(42)", None),
+        (
+            "aVariable.mockReturnValueOnce(Promise.resolve(42))",
+            "aVariable.mockResolvedValueOnce(42)",
+            None,
+        ),
+        (
+            "aVariable.mockReturnValueOnce(Promise.reject(42))",
+            "aVariable.mockRejectedValueOnce(42)",
+            None,
+        ),
+        // Todo: Fixed
+        // (
+        //     "aVariable.mockReturnValue(Promise.resolve({ target: 'world', message: 'hello' }))",
+        //     "aVariable.mockResolvedValue({ target: 'world', message: 'hello' })",
```

Comment:
Looks like our codegen has additional `\n\t` here, can we just put the first argument's source text here directly?(we can process this in the future PR)

Reply:
I've considered this before, but it will occur another problem in [this discussion](https://github.com/oxc-project/oxc/pull/2735#discussion_r1527029736), so I think I will spend some time to fix the codegen, and then back to fix this.

Reply:
> to fix the codegen

Add `\n\t` in our codegen is expected behavior, fix it may not a right direction.